ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java

```diff
@@ -5216,7 +5216,8 @@ private int updateFirstIncPendingFlag(Hive hive, ReplSetFirstIncLoadFlagDesc des
       for (String tableName : Utils.matchesTbl(hive, dbNameOrPattern, tableNameOrPattern)) {
         org.apache.hadoop.hive.metastore.api.Table tbl = hive.getMSC().getTable(dbNameOrPattern, tableName);
         parameters = tbl.getParameters();
-        if (ReplUtils.isFirstIncPending(parameters)) {
+        String incPendPara = parameters != null ? parameters.get(ReplUtils.REPL_FIRST_INC_PENDING_FLAG) : null;
+        if (incPendPara != null && (!flag.equalsIgnoreCase(incPendPara))) {
```

Comment:
If incPendPara = null also need to set the flag. Now it will skip.

Reply:
no ..as per current code ..we change this value only we are going to change it to other one ..else no need to change ..as this is called only after incremental is done